Week-4/Exercise4.1/index.js

```diff
@@ -0,0 +1,47 @@
+// Implement a function named getNumber which generates a random number.If random Number is divisible by 5 it will reject the promise else it will resolve the promise.Let’s also keep the promise resolution / rejection time as a variable.
+// 1.JS promises should not be used.
+// 2.A custom promise function should be created.
+// 3.This function should be able to handle all 3 states Resolve, Reject and Fulfilled.
+// 4.Should be able to accept callbacks as props.
+
+function getNumber(resolveCallback, rejectCallback, fulfillCallback) {
+  const randomNumber = Math.floor(Math.random() * 100) + 1;
+  let resolved = false;
+  let rejected = false;
+
+  const resolve = (value) => {
+    if (!resolved && !rejected) {
```

Comment:
why is this condition required

Reply:
This is to ensure whether promise is already resolved or not before attempting.